samples/javascript/chatapp-microsoft-entra-id/server.js

```diff
@@ -1,63 +1,153 @@
+import express from "express";
+import identity from "@azure/identity";
+
+import { WebPubSubServiceClient } from "@azure/web-pubsub";
+import { WebPubSubEventHandler } from "@azure/web-pubsub-express";
 
 const app = express();
-const hubName = 'sample_chat';
+const hubName = "sample_chat";
+
+let endpoint = process.argv[2]; // sample: https://<name>.webpubsub.azure.com
+
+const AuthType = Object.freeze({
+  AzurePowershell: 0,
+  VisualStudioCode: 1,
+  ApplicationWithClientSecret: 2,
+  ApplicationWithCertification: 3,
+  ApplicationWithFederatedIdentity: 4,
+  SystemAssignedMSI: 5,
+  UserAssignedMSI: 6,
+});
+
+var authType = AuthType.AzurePowershell;
+
+function GetTokenCredential(authType) {
+  switch (authType) {
+    case AuthType.AzurePowershell:
+      return GetAzurePowershellCredential();
+    case AuthType.VisualStudioCode:
+      return GetVisualStudioCodeCredential();
+    case AuthType.ApplicationWithClientSecret:
+      return GetApplicationWithClientSecretCredential();
+    case AuthType.ApplicationWithCertification:
+      return GetApplicationWithCertificationCredential();
+    case AuthType.ApplicationWithFederatedIdentity:
+      return GetApplicationWithFederatedIdentityCredential();
+    case AuthType.SystemAssignedMSI:
+      return GetSystemAssignedMSICredential();
+    case AuthType.UserAssignedMSI:
+      return GetUserAssignedMSICredential();
+    default:
+      throw new Error("Invalid auth type");
+  }
+}
+
+function GetAzurePowershellCredential() {
+  let options = new identity.AzurePowerShellCredentialOptions();
+  // options.AuthorityHost = AzureAuthorityHosts.AzureChina; // Entra ID China operated by 21Vianet
+  // options.AuthorityHost = AzureAuthorityHosts.AzureGovernment; // Entra ID US Government
+  options.AuthorityHost = identity.AzureAuthorityHosts.AzurePublicCloud;
+  return new identity.AzurePowerShellCredential(options);
+}
+
+function GetVisualStudioCodeCredential() {
+  let options = new identity.VisualStudioCodeCredentialOptions();
+  // options.AuthorityHost = AzureAuthorityHosts.AzureChina; // Entra ID China operated by 21Vianet
+  // options.AuthorityHost = AzureAuthorityHosts.AzureGovernment; // Entra ID US Government
+  options.AuthorityHost = identity.AzureAuthorityHosts.AzurePublicCloud;
+  return new identity.VisualStudioCodeCredential(options);
+}
+
+function GetApplicationWithClientSecretCredential() {
+  let options = new identity.ClientSecretCredentialOptions();
+  // options.AuthorityHost = AzureAuthorityHosts.AzureChina; // Entra ID China operated by 21Vianet
+  // options.AuthorityHost = AzureAuthorityHosts.AzureGovernment; // Entra ID US Government
+  options.authorityHost = identity.AzureAuthorityHosts.AzurePublicCloud;
+  return new identity.ClientSecretCredential("<tenantId>", "<clientId>", "<clientSecret>", options);
```

Comment:
we could use [dotenv](https://www.npmjs.com/package/dotenv) to read config values